zyte_common_items/components.py

```diff
@@ -148,3 +162,122 @@ class Metadata(Item):
     #: `0`. When there is no complete certainty, the value could be anything in
     #: between (e.g. `0.96`).
     probability: Optional[float] = None
+
+
+@attrs.define(kw_only=True)
+class BusinessPlaceMetadata(Metadata):
+    """Data extraction process metadata.
+
+    See :class:`BusinessPlace.metadata <zyte_common_items.items.BusinessPlace.metadata>`.
+    """
+
+    #: The search text the place was found with.
+    searchText: Optional[str] = None
+
+
+@attrs.define(kw_only=True)
+class Address(Item):
+    """Address item."""
+
+    #: The raw address information, as it appears on the website.
+    addressRaw: Optional[str] = None
+
+    #: The street address of the place.
+    streetAddress: Optional[str] = None
+
+    #: The city the place is located in.
+    addressCity: Optional[str] = None
+
+    #: The locality to which the place belongs.
+    addressLocality: Optional[str] = None
+
+    #: The region of the place.
+    addressRegion: Optional[str] = None
+
+    #: The country the place is located in.
+    #:
+    #: The country name or the two-letter ISO 3166-1 alpha-2 country code.
+    #: https://en.wikipedia.org/wiki/ISO_3166-1
+    addressCountry: Optional[str] = None
+
+    #: The postal code of the address.
+    postalCode: Optional[str] = None
+
+    #: The auxiliary part of the postal code.
+    #: It may include a state abbreviation or town name, depending on local standards.
+    postalCodeAux: Optional[str] = None
+
+    #: Geographical latitude of the place.
+    latitude: Optional[float] = None
+
+    #: Geographical longitude of the place.
+    longitude: Optional[float] = None
+
+
+@attrs.define(kw_only=True)
+class Amenity(Item):
+    """An amenity that a business place has"""
+
+    #: Name of amenity.
+    name: str
+
+    #: Availability of the amenity
+    value: bool
+
+
+@attrs.define(kw_only=True)
+class StarRating(Item):
+    """Official star rating of a place."""
+
+    #: Star rating of the place, as it appears on the page, without processing.
+    raw: Optional[str] = None
+
+    #: Star rating value of the place.
+    ratingValue: Optional[float] = None
+
+
+@attrs.define(kw_only=True)
+class ParentPlace(Item):
+    """If the place is located inside another place, these are the details of the parent place."""
+
+    #: Name of the parent place.
+    name: str
+
+    #: Identifier of the parent place.
+    placeId: str
+
+
+@attrs.define(kw_only=True)
+class OpeningHours(Item):
+    """specification of opening hours of a business place."""
+
+    #: English weekday name.
+    dayOfWeek: Optional[str] = attrs.field(
+        default=None,
+        validator=attrs.validators.optional(attrs.validators.in_(calendar.day_name)),
```

Comment:
From the docs of `day_name` (bold mine):

> An array that represents the days of the week in the **current locale**.

It may be best to follow the “don’t worry about validation for now” approach here as well. In fact, I wonder if you should actively remove all validation code altogether from this PR, otherwise we would have validation for some fields and not for others. Once we implement validation support, we should do it for all fields at once, rather than have partial validation now.

Reply:
Yes, will remove all validations.